src/DotNetWorker.Grpc/GrpcWorker.cs

```diff
@@ -235,10 +240,26 @@ internal static WorkerInitResponse WorkerInitRequestHandler(WorkerInitRequest re
             response.Capabilities.Add("RpcHttpTriggerMetadataRemoved", bool.TrueString);
             response.Capabilities.Add("UseNullableValueDictionaryForHttp", bool.TrueString);
             response.Capabilities.Add("TypedDataCollection", bool.TrueString);
+            response.Capabilities.Add("HandlesWorkerTerminateMessage", bool.TrueString);
 
             return response;
         }
 
+        internal static void WorkerTerminateRequestHandler(WorkerTerminate request)
+        {
+            // Terminate the worker process
+            try
+            {
+                var workerInformation = WorkerInformation.Instance;
+                var workerProcess = Process.GetProcessById(workerInformation.ProcessId);
+                workerProcess.Kill();
```

Comment:
Don't we also need to expose a Cancellation token instance to user's function code so that they can check the `IsCancellationRequested` property and act accordingly. May be it can be a property on the `FunctionContext` or a method parameter they can inject to the function method.

Reply:
`FunctionContext` is a type which carries data specific to a single invocation. Since the app shutdown (as in graceful shutdown) is not invocation specific, we probably do not need expose the cancellation token via FunctionContext. Users can inject `IApplicationLifeTime` to the their class constructor and listen via `ApplicationStopping` property which is a cancellation token, as needed.

I think we may need the Function level(via FunctionContext) cancellation when for the cancellation token feature @liliankasem  is driving.

Reply:
Updated the code based on our discussion offline.